containers.go

```diff
@@ -128,6 +129,10 @@ type rwContainerStore interface {
 	// Delete removes the record of the container.
 	Delete(id string) error
 
+	// DeferredDelete removes the record of the container, but does not remove data of container
+	// immediately. Data are moved in a temporary directory and removed in background.
+	DeferredDelete(id string) error
```

Comment:
In all cases: Please use package-private names for the methods on these package-private interfaces. They end up being private anyway (we just, in the past, didn’t change existing methods to minimize churn), and making that visible helps readers. (And if changing parameters of methods in these package-private interfaces, the methods can be renamed to package-private at the same time.)